### ISSUE TITLE
(5P) Help Button: Focus on Search Results when Searching Help Docs with Screen Reader

### ISSUE THREAD

**Issue by donalirl** [Automattic/wp-calypso#41508]
This is related to #41506

#### Steps to reproduce

1. Starting at WordPress.com while logged in, use a screen reader to navigate to the Help button
2. Tap the help button
3. Use a left to right swiping gesture to navigate through to the option to search the support docs
4. Type a search query such as Domains

#### What I expected

The voiceover cursor would focus on the search results, allowing me to navigate through and select the appropriate support article.

#### What happened instead

Voiceover cursor focuses on the entire browser window again. A visually-impaired user cannot access the search results of the search query they typed.

#### Browser / OS version
Chrome (latest), Windows 10
Chrome (latest) on android
Safari (latest) on iPhone

#### Screenshot / Video

Screen recording: https://d.pr/v/c8w5uZ

#### Context / Source

#p4vt7e-6n-p2

#journey #empathy #livesharing #accessibility #painpoint #user-report

**Comment by davemart-in** [comment]
@obenland and @jeryj perhaps this is something we could address with our current attention on the help container?

**Comment by jeryj** [comment]
@donalirl Could you re-test this? I merged in https://github.com/Automattic/wp-calypso/pull/43963 which changes the behavior of the search suggestions. I tested on VoiceOver in Safari iOS and it appears to be working. I don't have an Android device to test with.

**Comment by donalirl** [comment]
Hi @jeryj! It doesn't seem to be working with Android. After searching, the voiceover cursor focuses on the browser window, not the search results. I recorded that here: https://d.pr/v/1YYNdn

**Comment by jeryj** [comment]
Thanks for checking on this, @donalirl! 

It's surprising to me that it's moving focus from the search input. The code doesn't do anything to move the focus on search, but just loads new search results. Usually I see this kind of thing happen when the focused item is removed or becomes unfocusable, then the focus doesn't know where to go and resets to the top of the document/browser window. That isn't happening in this case, so I wonder what's causing it... 🤔

**Comment by sixhours** [comment]
This appears to be working OK for me; I'm not great with Voiceover, but I can tab through the site, access the help button, tab through the results, and open them and read them:

https://user-images.githubusercontent.com/2124984/128390090-295e1199-cd29-4b59-bb34-14ae70c5edfd.mov

I'm also able to do this on an iPhone with Voiceover enabled using taps/double-taps in Chrome.

I've no doubt there's more we can do for accessibility across Calypso! This appears to be working now, maybe as a result of similar work done in #44715, so I'm closing it out.

**Comment by msilbers** [comment]
> This appears to be working OK for me; I'm not great with Voiceover, but I can tab through the site, access the help button, tab through the results, and open them and read them:

@sixhours It looks like the comments above yours mention this may be specific to Android, is that what you tested in your screen recording?

**Comment by sixhours** [comment]
@msilbers Nope, I could not test on Android, only iPhone. Please feel free to reopen this if the issue still persists on Android and we can find someone to test it!

**Comment by msilbers** [comment]
Reopening as I was not able to get this working in Android

**Comment by mreishus** [comment]
We'll probably need to move this to another group. I don't have an android device; I spent some time trying to get an emulator running but wasn't successful.

**Comment by sixhours** [comment]
@yansern 👋 Does anyone in your group happen to have an Android device and could work on this?

**Comment by yansern** [comment]
Adding this to our inbox. Wondering if it's possible to test this with an emulator. Will see what we can do.

**Comment by donalirl** [comment]
Confirmed I can no longer recreate this! Seems fixed.

I tried to make a recording of it, but something seems to have changed with Android's screen reader - it doesn't seem compatible with the screen recording function anymore. So I wasn't able to get a recording, but I can't recreate the bug on the latest Android version.

**Comment by obenland** [comment]
Thank you!
